dspace-api/src/main/java/org/dspace/discovery/SolrServicePrivateItemPlugin.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.discovery;
+
+import static org.apache.logging.log4j.LogManager.getLogger;
+
+import java.sql.SQLException;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.core.Context;
+import org.dspace.core.LogManager;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * This plugin prevents discovery of private items by non-administrators.
+ */
+public class SolrServicePrivateItemPlugin implements SolrServiceSearchPlugin {
+
+    private static final Logger log = getLogger(SolrServicePrivateItemPlugin.class.getSimpleName());
+
+    @Autowired(required = true)
+    protected AuthorizeService authorizeService;
+
+
+    @Autowired(required = true)
+    protected SearchService searchService;
+
+    @Override
+    public void additionalSearchParameters(Context context, DiscoverQuery discoveryQuery, SolrQuery solrQuery) {
+        try {
+            // Prevents access if user has no administrative rights on the community or collection.
+            // NOTE: the resource restriction plugin adds location filters for community and collection admins.
+            if ( !authorizeService.isAdmin(context) && !authorizeService.isCommunityAdmin(context)
+                && !authorizeService.isCollectionAdmin(context)) {
+                solrQuery.addFilterQuery("NOT(discoverable:false)");
```

Comment:
Since items which are marked as not discoverable still have public read rights, this implies the SolrServiceResourceRestrictionPlugin would not filter them out (as you discovered in https://github.com/DSpace/DSpace/pull/2580#discussion_r353513281).
This change will imply that collection and community admins will find non-discoverable items with public read rights also outside their collection.
I don't think this is a blocking issue though because:
* This PR already ensures other users won't see the non-discoverable items in discovery
* If the items have read rights, it's probably not that big a deal for comm/coll admins to find them even if they're outside their collection